scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -584,16 +607,21 @@ else if(!analysisOnly &&  (missing instanceof MissingActivityTemplateConflict mi
             //decision-making here, we choose the first satisfying activity
             evaluation.forGoal(goal).associate(act, false);
             itConflicts.remove();
+            logger.info("Activity " + act + " has been associated to goal " + goal.getName() +" to satisfy conflict " + i);
             break;
+          } else{
+            logger.info("Activity " + act + " could not be associated to goal " + goal.getName() + " because of goal constraints");
           }
         }
       }
     }//for(missing)
 
 
     if(!missingConflicts.isEmpty() && goal.shouldRollbackIfUnsatisfied()){
+      logger.warn("Rolling back changes for "+goal.getName());
       rollback(goal);
     }
+    logger.info("Finishing goal satisfaction for goal " + goal.getName() +":"+ (missingConflicts.size() == 0 ? "SUCCESS" : "FAILURE") + ". Number of conflicts that could not be addressed: " + missingConflicts.size());
```

Comment:
Minor: Since `missingConflicts.size` will always be 0 in the `SUCCESS` case, merge reporting on the value with the `FAILURE` message
```suggestion
    logger.info("Finishing goal satisfaction for goal " + goal.getName() +":"+ (missingConflicts.size() == 0 ? "SUCCESS" : "FAILURE. Number of conflicts that could not be addressed: " + missingConflicts.size()));
```